### ISSUE TITLE
Style/BlockDelimiters should respect \.

### ISSUE THREAD

**Issue by envygeeks** [rubocop-hq/rubocop#3059]
Given the following code:

``` ruby
[].select { "Really Long" \
    }.each do |stuff|
end
```

Rubocop should respect the "\" and treat that as a single line just like Ruby does. That's what "\" is for.

**Comment by jonas054** [comment]
That's not valid Ruby. Can you rephrase your example code to make your point clear?

**Comment by envygeeks** [comment]
@jonas054 the example was clear and you should have been able to infer the missing each... it's been updated so that other bad programmers don't

**Comment by envygeeks** [comment]
Sorry, my bot is an ass, it's been disabled entirely.  It's going it again.  My bot is the one that filed the ticket.

**Comment by envygeeks** [comment]
It's been updated.

**Comment by tejasbubane** [comment]
@envygeeks Can you also report the following things:

* Expected behavior
* Actual behavior
* Steps to reproduce the problem
* RuboCop version

This helps us to reproduce and fix the problem easily. Thanks!

**Comment by envygeeks** [comment]
@tejasbubane it's literally all there in the ticket except for the version of which doesn't matter since we reproduced it on master (we only use Rubocop master.)

**Comment by alexdowad** [comment]
This totally makes sense. Off the top of my head, `ProcessedSource#lines` is the place to make the change. `Parser::Range` might also need to be tweaked.

**Comment by alexdowad** [comment]
Hmm. Having second thoughts on this one.

**Comment by envygeeks** [comment]
The example is bad, but it holds none the less, we had a line riding on our limit and that's how it folded.

**Comment by mikegee** [comment]
I implemented a fix in `ProcessedSource#lines` to resolve this, but `Commissioner#investigate` uses the AST, not the `#lines`. Also, I think such a change would cause a bunch of new `Metrics/LineLength` offenses for users.

Does anybody have an idea for a clean solution to this?

**Comment by bbatsov** [comment]
There hasn't been much activity on this ticket and our Core Team is spread too thin on so many tasks, so I'll just close it for the sake of having a cleaner lists of tasks to focus on. We'd gladly review a PR, but it's unlikely that'd we're going to tackle this ourselves in the foreseeable future.
